src/frontend/core/executor/impl/TriangleCountExecutor.cpp

```diff
@@ -34,13 +50,400 @@ TriangleCountExecutor::TriangleCountExecutor(SQLiteDBInterface *db, PerformanceS
     this->request = jobRequest;
 }
 
+static void allocate(int p, string w, std::map<int, string> &alloc, std::set<int> &remain,
+                     std::map<int, std::vector<string>> &p_avail, std::map<string, int> &loads) {
+    alloc[p] = w;
+    remain.erase(p);
+    p_avail.erase(p);
+    loads[w]++;
+    if (loads[w] >= 3) {
+        for (auto it = p_avail.begin(); it != p_avail.end(); it++) {
+            auto &ws = it->second;
+            auto itr = std::find(ws.begin(), ws.end(), w);
+            if (itr != ws.end()) {
+                ws.erase(itr);
+            }
+        }
+    }
+}
+
+static int get_min_partition(std::set<int> &remain, std::map<int, std::vector<string>> &p_avail) {
+    int p0 = *remain.begin();
+    size_t m = 1000000000;
+    for (auto it = remain.begin(); it != remain.end(); it++) {
+        int p = *it;
+        auto &ws = p_avail[p];
+        if (ws.size() > 0 && ws.size() < m) {
+            m = ws.size();
+            p0 = p;
+        }
+    }
+    return p0;
+}
+
+static const std::vector<int> LOAD_PREFERENCE = {2, 3, 1, 0};
+
+static int alloc_plan(std::map<int, string> &alloc, std::set<int> &remain, std::map<int, std::vector<string>> &p_avail,
+                      std::map<string, int> &loads) {
+    for (bool done = false; !done;) {
+        string w = "";
+        done = true;
+        int p;
+        for (auto it = remain.begin(); it != remain.end(); it++) {
+            p = *it;
+            if (p_avail[p].size() == 1) {
+                w = p_avail[p][0];
+                done = false;
+                break;
+            }
+        }
+        if (!w.empty()) allocate(p, w, alloc, remain, p_avail, loads);
+    }
+    if (remain.empty()) return 0;
+    int p0 = get_min_partition(remain, p_avail);
+    auto &ws = p_avail[p0];
+    if (ws.empty()) return (int)remain.size();
+    sort(ws.begin(), ws.end(), [&loads](string &w1, string &w2) {
+        return LOAD_PREFERENCE[loads[w1]] > LOAD_PREFERENCE[loads[w2]];
+    });  // load=1 goes first and load=3 goes last
```

Comment:
How about the other two values 0 and 2?

Reply:
The order is 1,0,2,3 for 4 cores

Reply:
Ok, lets update the comment with these details.

Reply:
Updated

Reply:
Ok